src/nuclide_header.F90

```diff
@@ -0,0 +1,820 @@
+    integer       :: zaid       ! ZAID identifier = 1000*Z + A
+    integer       :: filetype   ! ASCII or BINARY
+    integer       :: location   ! location of table within library
+    integer       :: recl       ! record length for library
+    integer       :: entries    ! number of entries per record
+    real(8)       :: awr        ! atomic weight ratio (# of neutron masses)
+    real(8)       :: kT         ! Boltzmann constant * temperature (MeV)
+    logical       :: metastable ! is this nuclide metastable?
+    character(MAX_FILE_LEN) :: path ! path to library containing table
+  end type XsListing
+
+  contains
+
+!===============================================================================
+! NUCLIDE_*_CLEAR resets and deallocates data in Nuclide_Base, Nuclide_Iso
+! or Nuclide_Angle
+!===============================================================================
+
+    subroutine nuclide_ce_clear(this)
+
+      class(Nuclide_CE), intent(inout) :: this ! The Nuclide object to clear
+
+      integer :: i ! Loop counter
+
+      if (associated(this % urr_data)) deallocate(this % urr_data)
+
+      if (allocated(this % reactions)) then
+        do i = 1, size(this % reactions)
+          call this % reactions(i) % clear()
+        end do
+      end if
+
+      call this % reaction_index % clear()
+
+    end subroutine nuclide_ce_clear
+
+    subroutine nuclide_iso_clear(this)
+
+      class(Nuclide_Iso), intent(inout) :: this ! The Nuclide object to clear
+
+      ! Cler the extended information
+      if (allocated(this % total)) then
+        deallocate(this % total, this % absorption, this % scatter)
+      end if
+      if (allocated(this % fission)) then
+        deallocate(this % fission, this % nu_fission)
+      end if
+      if (allocated(this % k_fission)) then
+        deallocate(this % k_fission)
+      end if
+      if (allocated(this % chi)) then
+        deallocate(this % chi)
+      end if
+      if (allocated(this % mult)) then
+        deallocate(this % mult)
+      end if
+
+    end subroutine nuclide_iso_clear
+
+    subroutine nuclide_angle_clear(this)
+
+      class(Nuclide_Angle), intent(inout) :: this ! The Nuclide object to clear
+
+      ! Cler the extended information
+      if (allocated(this % total)) then
+        deallocate(this % total, this % absorption, this % scatter)
+      end if
+      if (allocated(this % fission)) then
+        deallocate(this % fission, this % nu_fission)
+      end if
+      if (allocated(this % k_fission)) then
+        deallocate(this % k_fission)
+      end if
+      if (allocated(this % chi)) then
+        deallocate(this % chi)
+      end if
+
+      if (allocated(this % polar)) then
+        deallocate(this % polar)
+      end if
+      if (allocated(this % azimuthal)) then
+        deallocate(this % azimuthal)
+      end if
+      if (allocated(this % mult)) then
+        deallocate(this % mult)
+      end if
+
+    end subroutine nuclide_angle_clear
```

Comment:
Agree.  Thought about doing that now, but the base abstract Nuclide type should have a clear routine since the CE Nuclide type (Nuclide_CE) still needs it.  So these MG extended types have to implement something.  I agree that I dont need to do anything, but wouldnt it be a little odd just having a blank routine? 

That was my thinking at least